services/catalog/src/simcore_service_catalog/services/access_rights.py

```diff
@@ -0,0 +1,167 @@
+) -> Tuple[Optional[PositiveInt], List[ServiceAccessRightsAtDB]]:
+    """Given a service, it returns the owner's group-id (gid) and a list of access rights following
+    default access-rights policies
+
+    - DEFAULT Access Rights policies:
+        1. All services published in osparc prior 19.08.2020 will be visible to everyone (refered as 'old service').
+        2. Services published after 19.08.2020 will be visible ONLY to his/her owner
+        3. Front-end services are have read-access to everyone
+    """
+    db_engine: Engine = app.state.engine
+
+    groups_repo = GroupsRepository(db_engine)
+
+    everyone_gid = (await groups_repo.get_everyone_group()).gid
+    owner_gid = None
+    reader_gids: List[PositiveInt] = []
+
+    if _is_frontend_service(service) or await _is_old_service(app, service):
+        logger.debug("service %s:%s is old or frontend", service.key, service.version)
+        # let's make that one available to everyone
+        reader_gids.append(everyone_gid)
+
+    # try to find the owner
+    possible_owner_email = [service.contact] + [
+        author.email for author in service.authors
+    ]
+
+    for user_email in possible_owner_email:
+        possible_gid = await groups_repo.get_user_gid_from_email(user_email)
+        if possible_gid:
+            if not owner_gid:
+                owner_gid = possible_gid
+    if not owner_gid:
+        logger.warning("service %s:%s has no owner", service.key, service.version)
+    else:
+        reader_gids.append(owner_gid)
+
+    # we add the owner with full rights, unless it's everyone
+    default_access_rights = [
+        ServiceAccessRightsAtDB(
+            key=service.key,
+            version=service.version,
+            gid=gid,
+            execute_access=True,
+            write_access=(gid == owner_gid),
+            product_name=app.state.settings.access_rights_default_product_name,
+        )
+        for gid in set(reader_gids)
+    ]
+
+    # Patch releases inherit access rights from previous version
+
+    return (owner_gid, default_access_rights)
+
+
+async def evaluate_auto_upgrade_policy(
+    service_metadata: ServiceDockerData, services_repo: ServicesRepository
+):
+    # AUTO-UPGRADE PATCH policy:
+    #
+    #  - Any new patch released, inherits the access rights from previous compatible version
+    #  - TODO: add as option in the publication contract, i.e. in ServiceDockerData
+    #  - Does NOT apply to front-end services
+    #
+    # SEE https://github.com/ITISFoundation/osparc-simcore/issues/2244)
+    #
+    if _is_frontend_service(service_metadata):
+        return []
+
+    service_access_rights = []
+    new_version: Version = as_version(service_metadata.version)
+    latest_releases = await services_repo.list_service_releases(
+        service_metadata.key,
+        major=new_version.major,
+        minor=new_version.minor,
+        limit_count=1,
+    )
+    assert len(latest_releases) <= 1  # nosec
```

Comment:
ok so you can do that because you know the newest version is not yet in the database right? and the latest release will not return what service_metadata.version contains right?

Reply:
yes, but you have a good point ... this is wrong. In some situations it will return ``new_version``. Good catch!